CesiumGltfReader/test/TestGltfReader.cpp

```diff
@@ -332,7 +332,7 @@ TEST_CASE("Can deserialize KHR_draco_mesh_compression") {
 
   // Repeat test but this time the extension should be deserialized as a
   // JsonValue.
-  reader.getExtensions().setExtensionState(
+  reader.getOptions().setExtensionState(
```

Comment:
In this test, in theory the reader options could be set the same as the TilesetReader test does, `reader.getOptions().setCaptureUnknownProperties(true or false);`
From the looks of this PR, I assume that the GLTF reader is not supporting unknown properties?
If so, will the default state (true) of _captureUnknownProperties cause any problems?

Reply:
I'm not quite sure what you're asking here. `setCaptureUnknownProperties` should work on `GltfReader` the same as it does on `TilesetReader`. Unless I've missed something?

Reply:
Setting the capture flag on the options should work, yes. But for glTF reading, I didn't see a way to fetch the unknown properties object. 

This capture might show what I'm saying best...
![image](https://github.com/CesiumGS/cesium-native/assets/130494071/14e02b5d-845e-499a-9100-c049af9bdfc1)


Also, I didn't see a glTF reader test for unknown properties. For both these reasons, I assumed it wasn't supported.

Reply:
I still don't understand the problem. The glTF case is no different from the 3D Tiles one. Every generated class (including the glTF `Model` class) has an `unknownProperties` property. I added [two new tests](https://github.com/CesiumGS/cesium-native/pull/706/commits/2babed53b106290bc0906b1bd037e4d606412951) to demonstrate this, but it's definitely not worth adding such a test for _every_ generated class.

Reply:
You got it with the tests. It wasn't obvious to me how unknown properties were fetched with gltf results, but they are now. And there are tests for it too, nice.